samtranslator/model/sam_resources.py

```diff
@@ -1770,42 +1770,41 @@ class SamConnector(SamResourceMacro):
     """
 
     Source: Dict[str, Any]
-    Destination: Dict[str, Any]
+    Destination: Union[Dict[str, Any], List[Dict[str, Any]]]
     Permissions: List[str]
 
     resource_type = "AWS::Serverless::Connector"
     property_types = {
         "Source": PropertyType(True, dict_of(is_str(), any_type())),
-        "Destination": PropertyType(True, dict_of(is_str(), any_type())),
+        "Destination": PropertyType(
+            True, one_of(dict_of(is_str(), any_type()), list_of(dict_of(is_str(), any_type())))
+        ),
         "Permissions": PropertyType(True, list_of(is_str())),
     }
 
-    @cw_timer
-    def to_cloudformation(self, **kwargs: Any) -> List[Resource]:  # type: ignore
-        resource_resolver: ResourceResolver = kwargs["resource_resolver"]
-        original_template = kwargs["original_template"]
-
-        try:
-            destination = get_resource_reference(self.Destination, resource_resolver, self.Source)
-            source = get_resource_reference(self.Source, resource_resolver, self.Destination)
-        except ConnectorResourceError as e:
-            raise InvalidResourceException(self.logical_id, str(e))
-
+    def generate_resource(
+        self,
+        source: ConnectorResourceReference,
+        destination: ConnectorResourceReference,
+        dest_index: int,
+        multi_dest: bool,
+        resource_resolver: ResourceResolver,
+    ) -> List[Resource]:
         profile = get_profile(source.resource_type, destination.resource_type)
         if not profile:
             raise InvalidResourceException(
                 self.logical_id,
                 f"Unable to create connector from {source.resource_type} to {destination.resource_type}; it's not supported or the template is invalid.",
             )
 
+        profile_type, profile_properties = profile["Type"], profile["Properties"]
+
         # removing duplicate permissions
         self.Permissions = list(set(self.Permissions))
-        profile_type, profile_properties = profile["Type"], profile["Properties"]
```

Comment:
nit: should avoid unrelated changes (especially for cognitively high-load PRs like this), makes reviewing harder

Reply:
will update it.